torch/_dynamo/variables/lists.py

```diff
@@ -95,8 +95,18 @@ def call_method(
     ) -> "VariableTracker":
         options = VariableTracker.propagate(self, args, kwargs.values())
         if name == "__getitem__":
+            from .tensor import TensorVariable
+
             assert not kwargs and len(args) == 1
-            return self.getitem_const(args[0])
+            if isinstance(args[0], TensorVariable):
+                value = get_fake_value(args[0].as_proxy().node, tx)
```

Comment:
Why do we have a constant stored as a fake tensor?   Why isn't this just a constant?

Reply:
```list(range(5))[torch.as_tensor([1])]``` can pass in eager mode, that means 1 element scalar/constant tensor can be used as a python constant. In Dynamo, we'd like to know if this ```TensorVariable``` is a 1 element scalar/constant tensor, so we leverage```FakeTensor.constant``` to distinguish this. 
Do you suggest to add a ```constant``` attribute to ```TensorVariable``` and if ```TensorVariable.constant``` is not None, we can treat this tensor as constant in Dynamo?